src/tests.c

```diff
@@ -4361,11 +4365,17 @@ void test_xonly_pubkey_api(void) {
     CHECK(ecount == 2);
 
     ecount = 0;
-    CHECK(secp256k1_xonly_pubkey_parse(none, &pk, buf32) == 1);
     CHECK(secp256k1_xonly_pubkey_parse(none, NULL, buf32) == 0);
     CHECK(ecount == 1);
     CHECK(secp256k1_xonly_pubkey_parse(none, &pk, NULL) == 0);
     CHECK(ecount == 2);
+    /* Invalid field element */
+    CHECK(secp256k1_xonly_pubkey_parse(none, &pk, ones32) == 0);
+    CHECK(ecount == 2);
+    /* There's no point with x-coordinate 0 on secp256k1 */
+    CHECK(secp256k1_xonly_pubkey_parse(none, &pk, zeros32) == 0);
+    CHECK(ecount == 2);
+    CHECK(secp256k1_xonly_pubkey_parse(none, &pk, buf32) == 1);
```

Comment:
CHECK(ecount == 2); too?

Reply:
`ecount` can only be changed through illegal or error callbacks which should always entail return value of 0.